mlflow/utils/databricks_utils.py

```diff
@@ -1434,3 +1435,41 @@ def stage_model_for_databricks_model_serving(model_name: str, model_version: str
         },
     )
     augmented_raise_for_status(response)
+
+
+def databricks_api_disabled(api_name: str = "This API", alternative: str | None = None):
+    """
+    Decorator that disables an API method when used with Databricks.
+
+    This decorator checks if the tracking URI is a Databricks URI and raises an error if so.
+
+    Args:
+        api_name: Name of the API for the error message.
+        alternative: Optional alternative solution to suggest in the error message.
+
+    Returns:
+        Decorator function that wraps the method to check for Databricks.
+    """
+
+    def decorator(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
let's use `ParamSpec` here. A decorator like this erases type hints.

Reply:
Will update the util! Thanks for pointing this out @harupy